continuous_reporting/on_demand/dispatch

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash -ex
+
+# This is run from the ./launch script which sets LOG_FILE
+# and redirects all output from this script to it.
+
```

Comment:
If this is a shell script it would be good to use the `.sh` file extension?

Might also be good to have a comment at the top of the file to explain what this script does in 1 to 3 sentences.

Reply:
I expanded the description to further explain what it does and why it is used the way it is.

I don't usually put `.sh` extensions on executables because I think of it more as an implementation detail.
The path to the script is an entrypoint (like an API) but the language with which it is written (or whether its interpreted or compiled, etc) can be changed without breaking the API of whatever is calling it.

Reply:
I will leave the choice up to you, but imo it makes it more obvious that this is a script, and specifically a shell script, rather than something that was compiled into an executable, just by looking at which files are present in the repo.

Reply:
I'd rather have the repo more clear and more maintainable so I'm happy to rename them 👍 .
We can always insert a stub (to preserve the API) if we ever change the implementation.